test/Microsoft.AspNetCore.Server.KestrelTests/BadHttpRequestTests.cs

```diff
@@ -1,72 +1,99 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System;
 using System.Threading.Tasks;
 using Xunit;
 
 namespace Microsoft.AspNetCore.Server.KestrelTests
 {
     public class BadHttpRequestTests
     {
-        [Theory(Skip = "This test fails intermittently and needs to be investigated.")]
-        [InlineData("/ HTTP/1.1\r\n\r\n")]
-        [InlineData(" / HTTP/1.1\r\n\r\n")]
-        [InlineData("  / HTTP/1.1\r\n\r\n")]
-        [InlineData("GET  / HTTP/1.1\r\n\r\n")]
-        [InlineData("GET  /  HTTP/1.1\r\n\r\n")]
-        [InlineData("GET  HTTP/1.1\r\n\r\n")]
+        // Don't send more data than necessary to fail, otherwise the test might throw if
+        // it's still trying to send characters when the server had already shut down the connection
```

Comment:
@mikeharder I just did a quick test:

    const int serverPort = 8080;

    new Thread(_ =>
    {
        using (var server = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
        {
            server.Bind(new IPEndPoint(IPAddress.Loopback, serverPort));
            server.Listen(128);

            while (true)
            {
                using (var client = server.Accept())
                {
                    Console.WriteLine("Server: sending response");
                    client.Send(Encoding.ASCII.GetBytes("response"));
                    Console.WriteLine("Server: shutting down connection");
                    client.Shutdown(SocketShutdown.Both);
                }
            }
        }
    }).Start();

    using (var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
    {
        Console.WriteLine("Client: connecting");
        socket.Connect(IPAddress.Loopback, serverPort);
        Console.WriteLine("Client: sending request");
        socket.Send(Encoding.ASCII.GetBytes("request"));
        Thread.Sleep(1000);
        var responseBytes = new byte[1024];
        Console.WriteLine("Client: receiving response");
        socket.Receive(responseBytes);
        Console.WriteLine($"Response: {Encoding.ASCII.GetString(responseBytes)}");
    }

Output:

    Client: connecting
    Client: sending request
    Server: sending response
    Server: shutting down connection
    Client: receiving response

    Unhandled Exception: System.Net.Sockets.SocketException: An existing connection was forcibly closed by the remote host
       at System.Net.Sockets.Socket.Receive(Byte[] buffer, Int32 offset, Int32 size, SocketFlags socketFlags)
       at System.Net.Sockets.Socket.Receive(Byte[] buffer)
       at SocketTests.Program.Main(String[] args)

The `socket.Receive()` call at the end throws. I'm not 100% sure, but I think we should be able to read data available on a socket even if the remote end has closed the connection.